### PR TITLE
solution 2/3

### PR DESCRIPTION
**Pull request by igor-kashchenko**
[DEMO LINK](https://igor-kashchenko.github.io/react_todo-app-add-and-delete/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,155 @@
...
+      });
+  }, []);
+
+  const handleDeleteCompletedTodos = useCallback(() => {
+    todos.forEach(todo => {
+      if (todo.completed) {
+        handleDeleteTodo(todo.id);
+      }
+    });
+  }, []);
```
Comment (GarryRocksman) [inline src/App.tsx:116]: Here am not sure that all requests goes at the same time, i use Promise.all in my code

File: src/api/todos.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export const getTodos = (userId: number) => {
+  return client.get<Todo[]>(`/todos?userId=${userId}`);
+};
+
+export const addTodo = (userId: number, todo: Todo) => {
+  return client.post<Todo>(`/todos?userId=${userId}`, todo);
+};
+
+export const deleteTodo = (userId: number, todoId: number) => {
+  return client.delete(`/todos/${todoId}?userId=${userId}`);
```
Comment (GarryRocksman) [inline src/api/todos.ts:13]: here you dont need user Id 
Screenshot 2023-02-16 at 18 26 22

**Review by GarryRocksman — APPROVED** [review]
GJ🔥

File: src/components/Notification/Notification.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  onClear: () => void
+};
+
+export const Notification: React.FC<Props> = ({
+  hasError,
+  errorMessage,
+  onClear,
+}) => {
+  useEffect(() => {
+    setInterval(onClear, 3000);
```
Comment (Anna-Tkachenko) [inline src/components/Notification/Notification.tsx:16]: do you need to clear this interval?

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]
check the behavior on adding todo: when I add new todo, the loader should be visible only in component with temporary todo, not for all existing todos

**Review by denys-cheporniuk — APPROVED** [review]

**Pull request closed without merging**